SimplenoteUITests/NoteData.swift

```diff
@@ -2,7 +2,7 @@
 struct NoteData {
     let name: String
     let content: String
-    let tags: [String]
+    var tags: [String] = []
```

Comment:
That's a nice way to have the compiler generate the `init` for us 🙌 

One suggestion: make it `private` so the instance remains immutable for consumers, that is, they cannot change the value of `tags` after creating it.

```swift
private var tags: [String] = []
```

Reply:
>instance remains immutable for consumers

I did some playing around with this, and it looks like making `tags` variable `private` makes it impossible to create an instance in the first place, which _looks_ logical to me:

`'NoteData' initializer is inaccessible due to 'private' protection level`

So, it seems, that in order to achieve the instance immutability after creation, we really need to have all properties defined as constants, and have an `init()` with default value(s). (this is the approach you suggested intially)

Reply:
My bad 😞 . The right code was `private(set) var`... See #1247.

Reply:
No worries! :-) This made me learn/debug more, compared to the case if you have written `private(set)...` initially, so it's good